source/MapPanel.cpp

```diff
@@ -152,6 +152,51 @@ namespace {
 
 		return false;
 	}
+
+	bool BlinkMissionIndicator(const PlayerInfo &player, const Mission &mission, const int step)
```

Comment:
To have missions where the destination is farther than you can reach within the deadline display differently, we could perhaps have this function return a `pair<bool, int>`, where the first value is what the return value is now, and the second value is the number of spare days left. This function would be called before IsSatisfied is called, and if the number of spare days left is negative, isSatisfied would be false. So taking the two uses of this function, we'd see something like the following:

```
pair<bool, int> blink = BlinkMissionIndicator(player, mission, step)
if(!blink.first)
{
	bool isSatisfied = blink.second > 0 && IsSatisfied(player, mission);
	DrawPointer(from, missionCounter, isSatisfied ? currentColor : blockedColor, false);
}
```
```
pair<bool, int> blink = BlinkMissionIndicator(player, mission, step)
bool isSatisfied = blink.second > 0 && IsSatisfied(player, mission);
DrawPointer(system, it.drawn, blink.first ? black : isSatisfied ? currentColor : blockedColor, isSatisfied);
```